DevEdu.API/DevEdu.Business/Services/CourseService.cs

```diff
@@ -44,5 +47,49 @@ public List<CourseTopicDto> SelectAllTopicsByCourseId(int courseId)
             var list = _courseRepository.SelectAllTopicsByCourseId(courseId);
             return list;
         }
+        public void UpdateCourseTopicsByCourseId(int courseId, List<CourseTopicDto> topics)
+        {
+            CheckUniquenessPositions(topics);
+            var topicsInDatabase = _courseRepository.SelectAllTopicsByCourseId(courseId);
+            if (((topics.Count > topicsInDatabase.Count) || (topics.Count < topicsInDatabase.Count)) && topicsInDatabase.Count!=0)
+            {
+                DeleteAllTopicsByCourseId(courseId);
+                AddTopicsFromNewList(courseId, topics);
+            }
+            else if( topicsInDatabase.Count == 0)
+            {
+                AddTopicsFromNewList(courseId, topics);
+            }
+            else
+            {
+                foreach (var topic in topics)
+                {
+                    topic.Course = SetIdToCourseDto(courseId, topic.Course);
+                }
+                _courseRepository.UpdateCourseTopicsByCourseId(courseId, topics);
+            }
+        }
+        public void DeleteAllTopicsByCourseId(int courseId)
+        {
+            _courseRepository.DeleteAllTopicsByCourseId(courseId);
+        }
+        private void AddTopicsFromNewList(int courseId, List<CourseTopicDto> topics)
+        {
+            foreach (var topic in topics)
+            {
+                AddTopicToCourse(courseId, topic.Topic.Id, topic);
+            }
+        }
+        private void CheckUniquenessPositions(List<CourseTopicDto> topics)
+        {
+            if (topics.GroupBy(n => n.Position).Any(c => c.Count() > 1))
+            {
+                throw new Exception("the same positions of topics in the course");
+            }
```

Comment:
nice approach)
maybe it's better to add such checking for TopicIds?

Reply:
added, but maybe that's not what you wanted